### ISSUE TITLE
Labels not assigned to all the alerts when the EPS is high

### ISSUE THREAD

**Issue by odintree** [wazuh/wazuh#5461]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 3.12/3.13 | Wazuh component | Manager | Packages/Sources | OS version |

The labels do not assign to all the alerts from the group on high eps situations. 

**Reproduce**:
1) Added the `labels` to the group of the `agent`. 
```
<labels>
  <label key="customers">test</label>
</labels>
```
2) Artificially trigger the alerts on the `agent` side with the ~60 EPS. 
3) Result: 
The screenshot of the alerts where `labels` were assigned to the alerts: 
![image](https://user-images.githubusercontent.com/23520694/87164647-4e886c80-c2c9-11ea-841e-ffd56b1ce005.png)

The screenshot of the alerts from the same `agent` (and since it is the artificially triggered alerts, also with the same alerts) where there were no `labels`: 
![image](https://user-images.githubusercontent.com/23520694/87164863-9a3b1600-c2c9-11ea-8848-4388a20776a5.png)

 Could not get the same result when the EPS was at a lower rate. 

Kind regards,
Do Tuan Anh.

**Comment by snaow** [comment]
Hi, thanks for reporting.
We do have a couple of questions to start working on this issue:

- What EPS ratio do you think the manager starts losing the labels?
- Can we get more details about step two? How did you inject the events?

we would like you to confirm what of the following scenarios you think is happening:

1. Do the labels stop being populated after a specific moment and are not generated any more?
2. Are the labels not being populated at the beginning of the test for a period of time and after that period they start being generated properly with no issues?


Thanks.

**Comment by odintree** [comment]
Hi, 
Regarding the EPS rate, it is hard to say. But after some testing, I could get alerts without `labels` with ~1 EPS. It took approximately 25 minutes after I start a test. The lost period was for approximately 10 seconds. After that period of time, `labels` started to appear again. 
![image](https://user-images.githubusercontent.com/23520694/87699696-85f98c00-c795-11ea-994d-4e2e00272326.png)

For ~0.5 EPS I could not get any after a 1 hour testing time.
To reproduce this issue I use this script that I run on my `agent` machine:
```
#!/bin/bash
counter=1
while [ $counter -le 1000000 ]
do
  echo "Jul 10 14:57:23 opendistro yum[14247]: Installed: 2:vim-enhanced-7.4.629-6.el7.x86_64" >> /test/test.log
  echo $counter
  ((counter++))
  sleep .01
  done
echo All done
```
_Note: Log Collection is configured to monitor /test/test.log_

Regarding behavior:
As I mentioned above it took approximately 25 minutes to get the first alerts without a label. I got a test with higher EPS (~60) where I got the issue after 10 minutes, and then just got another test with the same EPS (~60) I got the issue only after 55 mins. So I could guess it works quite randomly. 
Usually, it happens for 10 seconds, and then it is going fine without any issue with labels. But it is repeatable as you could see in the screenshot below: 
![image](https://user-images.githubusercontent.com/23520694/87700396-8c3c3800-c796-11ea-8e6d-0cd8f574d3ee.png)

Another update: all the alerts that have no `labels`, also has no `agent.ip` field. 
![image](https://user-images.githubusercontent.com/23520694/87700738-1be1e680-c797-11ea-9ba2-f3d1c2621a68.png)


Kind regards,
Do Tuan Anh.

**Comment by vikman90** [comment]
## Issue reproduced

The manager enriches the alerts with the agents' labels using the agent-info files. Every 10 seconds (by default), the agents send a notification message containing the agent's version, the host platform, and the labels.

![diagram](https://user-images.githubusercontent.com/10536251/88292474-f6565f00-ccf9-11ea-82f6-df5a1656f206.png)

Remoted populates the agent-info file once every 10 seconds using the truncate & write method, so the file appears empty for a short time. That time span is negligible, but a high disk load could potentially increase it.

When Analysisd loads the agent-info file (either because the label data is not in the cache, or the cache is old) while it is being written, labels won't appear in any alert for that agent until the manager loads the file again.

## Possible solutions

1. Redesign the label data flow as we are performing in #5367, where the team is working to migrate the data stored in the agent-info files to a database managed by Wazuh DB.
2. Optimize the labels cache to detect this condition (agent-info file empty) and keep the old cached data.
3. Reducing the disk I/O load could help here as we determined that this issue is produced by disk stress.

**Comment by vikman90** [comment]
This issue has been fixed by #5367.
